fixieai/agents-ts/src/serve.ts

```diff
@@ -216,32 +224,28 @@ export default async function serve({
 
       const body = req.body as AgentResponse;
 
+      function serializedMessageOfMessage(message: Message): SerializedMessage {
+        const result: SerializedMessage = _.pick(message, 'text');
+        result.embeds = _.mapValues(message.embeds, (e) => e.serialize());
+        return result;
+      }
+
       try {
         const result = await funcHost.runFunction(funcName, body.message);
-        const response: AgentResponse = { message: { text: result } };
-        res.send(response);
+        const message = typeof result === 'string' ? { text: result } : result;
+        const agentResponse: AgentResponse = { message: serializedMessageOfMessage(message) };
+        res.send(agentResponse);
       } catch (e: any) {
         if (e.name === 'FunctionNotFoundError') {
           res.status(404).send(e.message);
           return;
         }
-
-        try {
-          const result = await funcHost.runFunction(funcName, body.message);
-          const response: AgentResponse = { message: { text: result } };
-          res.send(response);
-        } catch (e: any) {
-          if (e.name === 'FunctionNotFoundError') {
-            res.status(404).send(e.message);
-            return;
-          }
-          const errorForLogging = _.pick(e, 'message', 'stack');
-          logger.error(
-            { error: errorForLogging, functionName: funcName },
-            'Error running agent function',
-          );
-          res.status(500).send(errorForLogging);
-        }
```

Comment:
This was a merge error from before. (But all the unit tests still passed, so I guess it didn't impact correctness. 🤪 )

This section should just have been removed.